src/script/common/helper.cpp

```diff
@@ -54,17 +55,14 @@ template <> bool LuaHelper::readParam(lua_State *L, int index)
 	return lua_toboolean(L, index) != 0;
 }
 
-template <> bool LuaHelper::readParam(lua_State *L, int index, const bool &default_value)
+template <> s16 LuaHelper::readParam(lua_State *L, int index)
 {
-	if (lua_isnil(L, index))
-		return default_value;
-
-	return lua_toboolean(L, index) != 0;
+	return lua_tonumber(L, index);
```

Comment:
Possible improvement: Add range checks before blindly accepting over- or underflows.

Reply:
as this is not something I touched (and I'm also not sure how to), I'd rather leave it for another PR

Reply:
in that another PR it might be better to replace this call with `luaL_checkint`